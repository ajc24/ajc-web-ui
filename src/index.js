/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
