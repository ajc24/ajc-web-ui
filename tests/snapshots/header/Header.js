import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { Header } from '../../../src';
import testImage from '../../../stories/data/images/storybook-header-bg.png';

describe('Header', () => {
  let componentDidMountSpy;

  beforeAll(() => {
    componentDidMountSpy = jest
      .spyOn(Header.prototype, 'componentDidMount')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    componentDidMountSpy.mockRestore();
  });

  describe('Default props and rendering - Header with no top border and transparent background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <Header id="test-header-id">
            Header text content.
          </Header>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Header with grey top border and white background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <Header id="test-header-id" topBorder="grey" backgroundColour="white">
            Header text content.
          </Header>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Header with red top border and image background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <Header id="test-header-id" topBorder="red" imageData={testImage}>
            Header text content.
          </Header>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Header with no top border and grey background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <Header id="test-header-id" topBorder="none" backgroundColour="grey">
            Header text content.
          </Header>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Header with grey top border and yellow background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <Header id="test-header-id" topBorder="grey" backgroundColour="yellow">
            Header text content.
          </Header>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
