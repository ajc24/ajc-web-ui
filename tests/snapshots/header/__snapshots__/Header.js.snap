// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Default props and rendering - Header with no top border and transparent background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-center ajc-width-4k ajc-background-transparent"
  id="test-header-id-root"
  role="banner"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-header-content ajc-width-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-left ajc-width-hd-with-padding ajc-text-black ajc-font-default"
        id="test-header-id"
      >
        Header text content.
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Transferred props and rendering - Header with grey top border and white background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-center ajc-width-4k ajc-header-border-grey ajc-background-white"
  id="test-header-id-root"
  role="banner"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-header-content ajc-width-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-left ajc-width-hd-with-padding ajc-text-black ajc-font-default"
        id="test-header-id"
      >
        Header text content.
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Transferred props and rendering - Header with grey top border and yellow background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-center ajc-width-4k ajc-header-border-grey ajc-background-yellow"
  id="test-header-id-root"
  role="banner"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-header-content ajc-width-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-left ajc-width-hd-with-padding ajc-text-black ajc-font-default"
        id="test-header-id"
      >
        Header text content.
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Transferred props and rendering - Header with no top border and grey background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-center ajc-width-4k ajc-background-grey-1"
  id="test-header-id-root"
  role="banner"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-header-content ajc-width-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-left ajc-width-hd-with-padding ajc-text-black ajc-font-default"
        id="test-header-id"
      >
        Header text content.
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Transferred props and rendering - Header with red top border and image background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-center ajc-width-4k ajc-header-border-red ajc-background-transparent"
  id="test-header-id-root"
  role="banner"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-header-content ajc-width-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-left ajc-width-hd-with-padding ajc-text-black ajc-font-default"
        id="test-header-id"
      >
        Header text content.
      </div>
    </div>
  </div>
</div>
`;
